### PR TITLE
Enable a mechanism for filtering out log messages using

### LINKED ISSUE #1258: Create a way to control log message output.
Create a way to control log message output., most options should be passed via ConfigProto or another proto. Maybe add a LogOptions structure to hold options related to logging.  #1229
### Environment info

Operating System: Mac 10.9.5
### Feature request
1. Have a way to control logs that you record.As in gflags/re2 flags.Refer  #1229 
### What have you tried?
1. #1229

### PR DESCRIPTION
**Pull request by vrv**
an environment variable.  Reads the environment variable
on the first call to LOG, uses it to filter out based on the
severity of the log message.

By default, min log level is 0, mapping to INFO.
If you set the env var to 1, INFO is filtered out b t
WARNING still gets printed.

Parsing code is rudimentary, to avoid bringing in
dependencies into platform (e.g., strings/lib/numbers.h,
env_var.h, etc).  For this use-case it is probably fine.

Fixes #1258.

### CODE REVIEW

File: tensorflow/core/platform/default/logging.cc [hunk 1]
```diff
@@ -81,7 +82,38 @@ void LogMessage::GenerateLogMessage() {
...
+  const char* tf_env_var_val = getenv("TF_CPP_MIN_LOG_LEVEL");
+  if (tf_env_var_val == nullptr) {
+    return 0;
+  }
+
+  // Ideally we would use env_var / safe_strto64, but it is
+  // hard to use here without pulling in a lot of dependencies,
+  // so we do a poor-man's parsing.
+  StringPiece min_log_level_sp(tf_env_var_val);
+  string min_log_level_str(min_log_level_sp.data(), min_log_level_sp.size());
```
Comment (zheng-xq) [inline tensorflow/core/platform/default/logging.cc]: How about this? 

string min_log_level_str(tf_env_var_val)

File: tensorflow/core/platform/default/logging.cc [hunk 2]
```diff
@@ -81,7 +82,38 @@ void LogMessage::GenerateLogMessage() {
...
+  if (min_log_level_str == "1") {
```
Comment (zheng-xq) [inline tensorflow/core/platform/default/logging.cc]: How about also their actual names? For example, case insensitive, "info", "warning", "error", etc.

Comment (vrv) [inline tensorflow/core/platform/default/logging.cc]: info / warning /error don't indicate the order of INFO vs. WARNING and ERROR, so I think we should (for now) keep it as the integers.  I've added some comments at least.

**Review by zheng-xq — COMMENTED** [review]
LGTM

**Pull request merged**